iceoryx_hoofs/test/moduletests/test_posix_thread.cpp

```diff
@@ -35,64 +34,75 @@ class Thread_test : public Test
 
     void SetUp()
     {
-        m_run = true;
-        m_thread = new std::thread(&Thread_test::threadFunc, this);
     }
 
     void TearDown()
     {
-        m_run = false;
-        m_thread->join();
-        delete m_thread;
     }
 
     ~Thread_test()
     {
     }
 
-    void threadFunc()
-    {
-        while (m_run)
-        {
-            std::this_thread::sleep_for(std::chrono::milliseconds(10));
-        }
-    };
-
-    std::atomic_bool m_run{true};
-    std::thread* m_thread;
+    optional<Thread> sut;
 };
 
-#if !defined(__APPLE__)
+#if !defined(_WIN32) && !defined(__APPLE__)
+TEST_F(Thread_test, CreateThreadWithNonEmptyCallableSucceeds)
```

Comment:
Could you also add a tests where it is verified that the dtor calls join. It could look like.
1. Start a thread A which waits until thread B signals it that it goes out of scope
2. Start thread B whose callable blocks (you can yous `iceoryx_hoofs/testing/Barrier.hpp` for that)
3. Now let B go out of scope, this should be blocking so you start the time measurement directly before B goes out of scope.
4. Now let thread A wait  a given amount of time T and then signal the blocking callback that it can continue (the Barrier). Then thread B should be able to join.
5. Now measure the time, the time must be longer then T.

What we tested here is that the dtor of B blocks because of the join call. So the test could be called dtor of thread blocks until callback is finished.